docs/api/system-preferences.md

```diff
@@ -326,3 +325,19 @@ You can use the `setAppLevelAppearance` API to set this value.
 
 Sets the appearance setting for your application, this should override the
 system default and override the value of `getEffectiveAppearance`.
+
+### `systemPreferences.getMediaAccessStatus(mediaType)` _macOS_
+
+* `mediaType` String - `microphone` or `camera`.
+
+Returns `String` - `not-determined` | `granted` | `denied` | `restricted`
+
+This user consent was not required until macOS 10.14 Mojave, so this method will always return `granted` if your system is running 10.13 High Sierra or lower.
+
+### `systemPreferences.askForMediaAccess(mediaType)` _macOS_
+
+* `mediaType` String - the type of media being requested; can be `microphone`, `camera`, or `all` for all media to be requested.
+
+Returns `Promise<Boolean>` - A promise that resolves with `true` if consent was granted and `false` if it was denied. If an invalid `mediaType` is passed, the promise will be rejected. If an access request was denied and later is changed through the System Preferences pane, a restart of the app will be required for the new permissions to take effect. If access has already been requested and denied, it _must_ be changed through the preference pane; an alert will not pop up and the promise will resolve with the existing access status.
```

Comment:
How does this behave in the case that `all` was requested, and only one of {`microphone`, `camera`} was granted?

Reply:
> How does this behave in the case that `all` was requested, and only one of {`microphone`, `camera`} was granted?

I could be wrong here, but it looks like the appropriate values for microphone access and camera access will be properly set in the access controller, but the promise will resolve to the value of `self.hasFullMediaAccess`, which would be `false` if one of the two types of access was not granted (even if the other was).

A couple ideas:

1. Don't return a `Promise<Boolean>` from `systemPreferences.askForMediaAccess`, but instead just a `Promise<null>` or something similar, and require the user to specifically call `getMediaAccessStatus("microphone")` or `getMediaAccessStatus("camera")` to get resulting value.

2. Introduce a `MediaPermissions` type with the keys `{ microphone: Boolean, camera: Boolean }` and return `Promise<MediaPermissions>` from `systemPreferences.askForMediaAccess`. (Perhaps `getMediaAccessStatus` could just take no arguments in this case and return the `MediaPermissions` object for symmetry.)

I kinda like option 2, curious if others have thoughts. 🤔💭

Reply:
I'm a fan of (3): don't expose the `all` option at all :)

or, if we must have it, then (4): return `false` if either permission was denied and let the user check specifically what was denied using `getMediaAccessStatus`